src/bvals/cc/sparse_bvals_cc_in_one.cpp

```diff
@@ -0,0 +1,358 @@
+//========================================================================================
+// Parthenon performance portable AMR framework
+// Copyright(C) 2020 The Parthenon collaboration
+// Licensed under the 3-clause BSD License, see LICENSE file for details
+//========================================================================================
+// (C) (or copyright) 2022. Triad National Security, LLC. All rights reserved.
+//
+// This program was produced under U.S. Government contract 89233218CNA000001 for Los
+// Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+// for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+// in the program are reserved by Triad National Security, LLC, and the U.S. Department
+// of Energy/National Nuclear Security Administration. The Government is granted for
+// itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do so.
+//========================================================================================
+
+#include <algorithm>
+#include <iostream> // debug
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "bvals_cc_in_one.hpp"
+#include "config.hpp"
+#include "globals.hpp"
+#include "interface/variable.hpp"
+#include "kokkos_abstraction.hpp"
+#include "mesh/mesh.hpp"
+#include "mesh/mesh_refinement.hpp"
+#include "mesh/meshblock.hpp"
+#include "mesh/refinement_cc_in_one.hpp"
+#include "utils/error_checking.hpp"
+
+namespace parthenon {
+namespace cell_centered_bvars {
+
+namespace detail { 
+template<class F>
+void IterateBoundaries(std::shared_ptr<MeshData<Real>> &md, F func) {
+  for (int block = 0; block < md->NumBlocks(); ++block) {
+    auto &rc = md->GetBlockData(block);
+    auto pmb = rc->GetBlockPointer();
+    for (auto &v : rc->GetCellVariableVector()) {
+      if (v->IsSet(Metadata::FillGhost)) {
+        for (int n = 0; n < pmb->pbval->nneighbor; ++n) {
+          auto& nb = pmb->pbval->neighbor[n];
+          func(pmb, rc, nb, v);
+        }
+      }
+    }
+  }
+}
```

Comment:
Should we use this in `bvals_in_one` too? if so maybe it should be moved to somewhere that can be shared accross `bvals` stuff.

Reply:
My thinking was that this would eventually replace `bvals_in_one` (since it basically does the same thing). So, I didn't think it was worth it to make changes to `bvals_in_one`.